### DISCUSSION TITLE
Why not use Microsoft.Extensions.DependencyInjection?

### DISCUSSION THREAD

**Discussion by jakenuts** [spectreconsole/spectre.console#1395] [Q&A] [answered]
I understand there might have been cause to work with an internally developed IoC provider back in the early days of .NET Core but as Microsoft.Extensions.DependencyInjection is so ubiquitous in apps on this platform I was wondering why TypeRegistrar is still necessary. It always feels a bit off to use ServiceCollection for all the primary setup and then hand it off to a different class with (likely) it's own behaviors.

**Comment by patriksvensson** [comment, marked as answer]
It's an adapter. We don't wish to have a dependency on a specific IoC container.
